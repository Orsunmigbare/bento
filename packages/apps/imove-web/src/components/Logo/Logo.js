import React from "react"
import styled from "styled-components"
import { color } from "@staccx/theme"

const Logo = ({ variant, width, height }) => (
  <div>
    <SVG
      width={width}
      height={height}
      viewBox="0 0 110 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <Letters variant={variant}>
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M95.0104 19.2541C99.0444 19.2114 106.867 19.1048 106.867 15.5821C106.867 13.9808 105.188 12.9987 102.956 12.9987C99.0857 12.9987 96.3208 15.7955 95.0104 19.2541ZM94.2936 22.2004C94.2324 22.7341 94.1914 23.2467 94.1914 23.7589C94.1914 26.9187 95.6248 28.712 99.0037 28.712C100.937 28.712 102.571 28.1043 104.172 26.9231C104.822 26.4439 105.716 26.5619 106.235 27.1904C106.828 27.909 106.708 29.0126 105.963 29.5571C103.669 31.2359 101.352 32.0002 98.676 32.0002C93.1264 32.0002 90.6895 28.7547 90.6895 23.8017C90.6895 16.7348 95.6658 9.9458 103.058 9.9458C107.727 9.9458 110.001 12.6571 110.001 15.6462C110.001 20.9622 102.69 22.2215 94.2936 22.2004Z"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M71.1934 10.3096C73.4817 10.3096 75.5303 11.7885 76.3356 14.0217L80.8686 26.5916C80.9289 26.7588 81.1549 26.7588 81.2156 26.5916L85.7646 14.0199C86.5724 11.7873 88.6218 10.3096 90.9107 10.3096L83.0544 30.3158C82.72 31.1673 81.9251 31.7232 81.0424 31.7232C80.1593 31.7232 79.3638 31.1664 79.0297 30.314L71.1934 10.3096Z"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M67.2301 20.9194C67.2301 16.6283 64.9366 13.0842 60.7179 13.0842C56.5199 13.0842 54.2467 16.6066 54.2467 20.9194C54.2467 25.2536 56.5199 28.8616 60.7179 28.8616C64.9366 28.8616 67.2301 25.2323 67.2301 20.9194ZM50.6426 20.9194C50.6426 14.8561 54.7178 9.9458 60.7179 9.9458C66.7179 9.9458 70.8344 14.8561 70.8344 20.9194C70.8344 27.0255 66.7179 32.0002 60.7179 32.0002C54.7178 32.0002 50.6426 27.0255 50.6426 20.9194Z"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M40.1904 9.9458C37.3121 9.9458 34.8244 11.3389 33.4673 14.3614C32.4599 11.2104 30.0132 9.9458 27.2376 9.9458C24.76 9.9458 22.5766 11.0026 21.3609 13.4621C21.2676 12.7639 20.9891 12.1284 20.5741 11.6138C19.9833 10.8217 19.0632 10.31 18.0263 10.31H17.985C17.9508 10.3091 17.9173 10.3047 17.8828 10.3047V29.8914C17.8828 30.7775 18.4862 31.5166 19.2881 31.687C19.4016 31.7113 19.5186 31.7236 19.6388 31.7236H19.661C20.6207 31.7236 21.3985 30.913 21.3985 29.9125V19.3346C21.3985 15.9047 23.3516 13.3324 26.5593 13.3324C29.6636 13.3324 30.8973 15.6046 30.8973 18.9912V29.9125C30.8973 30.913 31.6751 31.7236 32.6345 31.7236C33.5942 31.7236 34.372 30.913 34.372 29.9125V19.3346C34.372 15.9047 36.3459 13.3324 39.512 13.3324C42.7194 13.3324 43.8706 15.6904 43.8706 19.5272V29.9125C43.8706 30.913 44.6484 31.7236 45.6081 31.7236H45.6286C46.5883 31.7236 47.366 30.913 47.366 29.9125V19.0987C47.366 13.9328 45.2071 9.9458 40.1904 9.9458Z"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M9.88477 10.3047V29.8914C9.88477 30.9034 10.6718 31.7236 11.6427 31.7236C12.6134 31.7236 13.4004 30.9034 13.4004 29.8914V13.97C13.4004 11.9457 11.8264 10.3047 9.88477 10.3047Z"
        />
      </Letters>
      <Blob
        variant={variant}
        fillRule="evenodd"
        clipRule="evenodd"
        d="M3.51342 7.32591H7.03162V3.66296C7.03162 1.63725 5.45643 0 3.51342 0C1.57491 0 0 1.63725 0 3.66296C0 4.67815 0.396183 5.59147 1.02991 6.25217C1.66335 6.91784 2.54416 7.32591 3.51342 7.32591Z"
      />
      <defs>
        <linearGradient
          id="gradient"
          x1="2.38665"
          y1="8.78196"
          x2="0.264382"
          y2="1.0677"
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#92F8B8" />
          <stop offset="1" stopColor="#1D966C" />
        </linearGradient>
      </defs>
    </SVG>
  </div>
)

const SVG = styled.svg`
  width: ${p => p.width};
  height: ${p => p.height};
`

const Letters = styled.g`
  fill: ${p => (p.variant === "1" ? color("secondary") : color("white"))};
`

const Blob = styled.path`
  fill: ${p => (p.variant === "1" ? `url(#gradient)` : color("white"))};
`

export default Logo
