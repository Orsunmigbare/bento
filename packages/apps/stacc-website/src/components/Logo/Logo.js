import React from "react"
import styled from "styled-components"

const Logo = ({ inverted }) => (
  <SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 158 50">
    <g fill="none" fillRule="evenodd">
      <g transform="translate(2 2)">
        <polygon
          stroke={!inverted ? "#E6503F" : "#ffffff"}
          strokeWidth="3.5"
          points="0 11.503 0 34.497 19.462 46 38.923 34.497 38.923 11.503 19.462 0"
        />
        <path
          stroke={!inverted ? "#E6503F" : "#ffffff"}
          strokeWidth="3.5"
          d="M0 11.5L19.46 23m19.46-11.5L19.46 23m0 23V23M29.2 5.75L9.73 17.25M0 23l19.46 11.5m9.74 5.75v-23"
        />
        <path
          fill={!inverted ? "#2b2b2b" : "#ffffff"}
          fillRule="nonzero"
          d="M62.56 34c-1.64 0-3.25-.22-4.81-.67a11.87 11.87 0 0 1-3.75-1.7l1.82-3.75c.9.64 2 1.15 3.26 1.54 1.28.37 2.55.56 3.78.56 2.83 0 4.24-.78 4.24-2.33 0-.74-.36-1.25-1.1-1.54a17.5 17.5 0 0 0-3.44-.83 27.84 27.84 0 0 1-4.1-.9 6.44 6.44 0 0 1-2.72-1.78 5.28 5.28 0 0 1-1.13-3.6c0-2 .8-3.58 2.38-4.77 1.62-1.2 3.79-1.81 6.52-1.81 1.38 0 2.77.17 4.16.51 1.4.32 2.53.75 3.41 1.3l-1.82 3.75a10.87 10.87 0 0 0-5.8-1.58c-1.38 0-2.44.23-3.17.67-.71.43-1.06 1-1.06 1.7 0 .79.38 1.36 1.13 1.7.79.31 1.98.62 3.6.9 1.62.27 2.94.57 3.98.91 1.03.34 1.92.92 2.65 1.74a4.91 4.91 0 0 1 1.13 3.47c0 1.97-.82 3.55-2.46 4.73-1.64 1.19-3.87 1.78-6.7 1.78zm24.6-1.42c-.55.47-1.23.83-2.04 1.06a8.6 8.6 0 0 1-2.5.36c-2.22 0-3.94-.6-5.15-1.81-1.21-1.21-1.82-2.98-1.82-5.3V8h4.73v4.81h5.42v3.95h-5.42v10.02c0 1.03.24 1.81.72 2.37.5.52 1.2.79 2.09.79 1.06 0 1.94-.3 2.65-.87l1.32 3.51zm23.02-19.92v21.06h-4.5V31a7.33 7.33 0 0 1-2.92 2.25 9.3 9.3 0 0 1-3.79.75 10.5 10.5 0 0 1-5.3-1.34 9.53 9.53 0 0 1-3.64-3.79 12.02 12.02 0 0 1-1.29-5.68c0-2.16.43-4.04 1.3-5.64a9.53 9.53 0 0 1 3.63-3.79 10.5 10.5 0 0 1 8.9-.63 7.91 7.91 0 0 1 2.88 2.09v-2.56h4.73zM99.5 29.78c1.74 0 3.18-.6 4.32-1.82a6.69 6.69 0 0 0 1.7-4.77c0-1.97-.56-3.56-1.7-4.77a5.67 5.67 0 0 0-4.32-1.82c-1.74 0-3.18.6-4.32 1.82a6.78 6.78 0 0 0-1.66 4.77c0 1.97.55 3.56 1.66 4.77a5.67 5.67 0 0 0 4.32 1.82zM125.65 34c-2.1 0-3.97-.46-5.64-1.38a9.95 9.95 0 0 1-3.9-3.83 11.2 11.2 0 0 1-1.4-5.6 11 11 0 0 1 1.4-5.56c.93-1.63 2.22-2.91 3.86-3.83a11.55 11.55 0 0 1 5.68-1.38c2 0 3.74.42 5.23 1.26a8.02 8.02 0 0 1 3.4 3.63l-3.63 2.21a5.68 5.68 0 0 0-2.2-2.17c-.85-.5-1.8-.75-2.83-.75-1.77 0-3.24.6-4.4 1.82a6.52 6.52 0 0 0-1.74 4.77c0 2 .57 3.6 1.7 4.81a5.95 5.95 0 0 0 4.44 1.78c1.03 0 1.98-.24 2.84-.71a5.99 5.99 0 0 0 2.2-2.21l3.63 2.2a8.47 8.47 0 0 1-3.45 3.68 10.37 10.37 0 0 1-5.19 1.26zm21.72 0c-2.1 0-3.98-.46-5.65-1.38a9.95 9.95 0 0 1-3.9-3.83 11.21 11.21 0 0 1-1.4-5.6 11 11 0 0 1 1.4-5.56c.94-1.63 2.22-2.91 3.86-3.83a11.55 11.55 0 0 1 5.69-1.38c1.99 0 3.73.42 5.22 1.26a8.02 8.02 0 0 1 3.41 3.63l-3.64 2.21a5.7 5.7 0 0 0-2.2-2.17c-.85-.5-1.8-.75-2.83-.75-1.77 0-3.24.6-4.4 1.82a6.52 6.52 0 0 0-1.74 4.77c0 2 .57 3.6 1.7 4.81a5.94 5.94 0 0 0 4.44 1.78c1.03 0 1.98-.24 2.84-.71a6 6 0 0 0 2.2-2.21l3.63 2.2a8.48 8.48 0 0 1-3.45 3.68 10.37 10.37 0 0 1-5.19 1.26z"
        />
      </g>
    </g>
  </SVG>
)

const SVG = styled.svg`
  width: 158px;
  height: 50px;
`

export default Logo
