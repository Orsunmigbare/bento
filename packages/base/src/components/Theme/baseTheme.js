export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
