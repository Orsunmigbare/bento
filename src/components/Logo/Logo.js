import PropTypes from "prop-types"
import React from "react"

const Logo = ({ brand, height, width, className }) => (
  <img
    height={height}
    width={width}
    src={`https://api.logos.staccsandbox.com/${brand}`}
    alt={brand}
    className={className}
  />
)

export default Logo

Logo.propTypes = {
  brand: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number
}

Logo.defaultProps = {
  height: 48
}
