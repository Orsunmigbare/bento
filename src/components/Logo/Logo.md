```jsx
  <Logo brand={"kiwi"} token={"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1YWUxODg2MDE1MDc0MDAwNDNjZDRlZTYiLCJlbWFpbCI6ImluZm9AZmluYW5jZWlubm92YXRpb24ubm8iLCJyb2xlIjoidXNlciIsImlhdCI6MTUyODcwMjIyNCwiZXhwIjoxNTM3MzQyMjI0LCJhdWQiOiJTdGFjY1g6Q2xpZW50cyIsImlzcyI6IlN0YWNjOlN0YWNjWCJ9.qrlm-hpNK8nYdBpNGswdX0uRlzpsH4Okv6VbwmReeOQ"} />
  ```

```jsx
    <Logo brand={"sbanken"} token={"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1YWUxODg2MDE1MDc0MDAwNDNjZDRlZTYiLCJlbWFpbCI6ImluZm9AZmluYW5jZWlubm92YXRpb24ubm8iLCJyb2xlIjoidXNlciIsImlhdCI6MTUyODcwMjIyNCwiZXhwIjoxNTM3MzQyMjI0LCJhdWQiOiJTdGFjY1g6Q2xpZW50cyIsImlzcyI6IlN0YWNjOlN0YWNjWCJ9.qrlm-hpNK8nYdBpNGswdX0uRlzpsH4Okv6VbwmReeOQ"} />
```