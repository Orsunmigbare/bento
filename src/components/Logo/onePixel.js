export default "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAABGdBTUEAALGPC/xhBQAAAAtJREFUCB1jYAACAAAFAAGNu5vzAAAAAElFTkSuQmCC"
