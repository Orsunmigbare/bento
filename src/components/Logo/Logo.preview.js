import React from "react"
import Logo from "./Logo"
import Wrapper from "../Layout/Wrapper/Wrapper"
import Button from "../Button/Button"
import Input from "../Forms/Input/Input"

const preview = {
  title: "Logo",
  category: "Components/Dataviz",
  component: Logo,
  render: props => (
    <React.Fragment>
      <Logo
        brand={"kiwi"}
        token={
          "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1YWUxODg2MDE1MDc0MDAwNDNjZDRlZTYiLCJlbWFpbCI6ImluZm9AZmluYW5jZWlubm92YXRpb24ubm8iLCJyb2xlIjoidXNlciIsImlhdCI6MTUyODcwMjIyNCwiZXhwIjoxNTM3MzQyMjI0LCJhdWQiOiJTdGFjY1g6Q2xpZW50cyIsImlzcyI6IlN0YWNjOlN0YWNjWCJ9.qrlm-hpNK8nYdBpNGswdX0uRlzpsH4Okv6VbwmReeOQ"
        }
        renderFallback={() => (
          <img
            src={"http://fillmurray.com/100/100"}
            alt={`${props.brand || "Bill Murray"}`}
          />
        )}
        {...props}
      />
    </React.Fragment>
  ),
  renderExampleController: ({ setComponentState, search }) => (
    <Wrapper>
      <Input
        id={"logoSearch"}
        onChange={e => setComponentState({ search: e.target.value })}
      />
      <Button onClick={() => setComponentState({ brand: search })}>
        Search
      </Button>
    </Wrapper>
  )
}

export default preview
