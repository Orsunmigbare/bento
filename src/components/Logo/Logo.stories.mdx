import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Logo from "./Logo"

<Meta title="Components|Logo" component={Logo}/>

# Logo
Forklaring p√• hva det er og hvordan den kan brukes

```javascript
import { Logo } from "@staccx/bento"
```

## Standard Logo
This is the standard Logo

<Preview>
  <Story name="Standard">
   <Logo brand={"dnb"} />
  </Story>
</Preview>

## Flere stories her
Hva er spesielt med denne storien?

## Props for Logo
<Props of={Logo} />
