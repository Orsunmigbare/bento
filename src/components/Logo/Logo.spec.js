import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Logo from "./Logo"

describe("Logo", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Logo />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Height", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Logo height={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("width", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Logo width={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("brand", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Logo brand="dnb" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
