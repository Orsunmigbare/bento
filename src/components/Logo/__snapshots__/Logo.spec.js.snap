// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Snapshots Height 1`] = `
<img
  height={100}
  src="https://api.logos.staccsandbox.com/undefined"
/>
`;

exports[`Logo Snapshots No props 1`] = `
<img
  height={48}
  src="https://api.logos.staccsandbox.com/undefined"
/>
`;

exports[`Logo Snapshots brand 1`] = `
<img
  alt="dnb"
  height={48}
  src="https://api.logos.staccsandbox.com/dnb"
/>
`;

exports[`Logo Snapshots width 1`] = `
<img
  height={48}
  src="https://api.logos.staccsandbox.com/undefined"
  width={100}
/>
`;
